test/runtests.jl

```diff
@@ -23,14 +23,44 @@ end
 
 linreg(x::AbstractVecOrMat, y::AbstractVector) = qr!(simplemm(x)) \ y
 
+@testset "Test GLmResp/LmResp constructor" begin
+    y = ifelse.(randn(10) .< 0, 1.0, 0.0)
+    d = Binomial()
+    η = similar(y)
+    μ = similar(y)
+    wts = similar(y)
+    aw = aweights(wts)
+    uw = GLM.uweights(10)
+    o = ones(10)
+    oe = Float64[]
+    l = GLM.LogitLink()
+
+    ## wts must be AbstractWeights
+    @test_throws ArgumentError GLM.GlmResp(y, d, l, η, μ, oe, wts)
+    ## Length of weights must match length of y
+    @test_throws DimensionMismatch GLM.GlmResp(y, d, l, η, μ, oe, uw[1:9])
+    @test_throws DimensionMismatch GLM.LmResp{typeof(y),typeof(aw)}(μ, oe, aw[1:2], y)
+    @test_throws DimensionMismatch GLM.LmResp{typeof(y),typeof(aw)}(μ, oe, aw, y[1:2])
+    ## Length of y must match length of η
+    @test_throws DimensionMismatch GLM.GlmResp(y[1:9], d, l, η, μ, oe, uw)
+    ## Length of y must match length of μ
+    @test_throws DimensionMismatch GLM.GlmResp(y, d, l, η, μ[1:9], oe, uw)
+    @test_throws DimensionMismatch GLM.LmResp{typeof(y),typeof(aw)}(μ[1:2], oe, aw, y)
+    ## Length of offset must match length of y
+    @test_throws DimensionMismatch GLM.GlmResp(y, d, l, η, μ, o[1:9], uw)
+    @test_throws DimensionMismatch GLM.LmResp{typeof(y),typeof(aw)}(μ, o[1:9], aw, y)
+    ## y must be in the support of d
+    @test_throws ArgumentError GLM.GlmResp([1.0, 2.0], d, l, η, μ, o, uw)
+end
+
 @testset "LM with $dmethod" for dmethod in (:cholesky, :qr)
     Σ = [6.136653061224592e-05 -9.464489795918525e-05
          -9.464489795918525e-05 1.831836734693908e-04]
 
     lm1 = fit(LinearModel, @formula(OptDen ~ Carb), form; method=dmethod)
     test_show(lm1)
     @test isapprox(coef(lm1), linreg(form.Carb, form.OptDen))
-
+    @test residuals(lm1.rr) ≈ residuals(lm1)
```

Comment:
This doesn't seem to test much that the behavior is correct. How about checking against actual values? (Possibly only a subset of them if the vector is too long.)

Same below.

Reply:
Still applies?